File: python/batch_job.py
import json
import os
import sys
import urllib.request

API_BASE = os.environ.get("TRANSCRIBETOOL_BASE_URL", "https://transcribetool.com")
API_KEY = os.environ.get("TRANSCRIBETOOL_API_KEY")

if not API_KEY:
    raise RuntimeError("Set TRANSCRIBETOOL_API_KEY")

input_text = sys.argv[1] if len(sys.argv) > 1 else "https://youtube.com/playlist?list=YOUR_PLAYLIST_ID"

for path, payload in [
    ("/api/v1/batch-jobs/estimate", {"input_text": input_text}),
    ("/api/v1/batch-jobs", {"source_app": "github_example", "input_text": input_text}),
]:
    req = urllib.request.Request(
        f"{API_BASE}{path}",
        data=json.dumps(payload).encode("utf-8"),
        headers={
            "x-api-key": API_KEY,
            "content-type": "application/json",
        },
        method="POST",
    )
    with urllib.request.urlopen(req) as resp:
        print(path)
        print(resp.read().decode("utf-8"))

File: javascript/batch-job.mjs
const API_BASE = process.env.TRANSCRIBETOOL_BASE_URL || "https://transcribetool.com";
const API_KEY = process.env.TRANSCRIBETOOL_API_KEY;

if (!API_KEY) {
  throw new Error("Set TRANSCRIBETOOL_API_KEY");
}

const inputText = process.argv[2] || "https://youtube.com/playlist?list=YOUR_PLAYLIST_ID";

const estimate = await fetch(`${API_BASE}/api/v1/batch-jobs/estimate`, {
  method: "POST",
  headers: {
    "x-api-key": API_KEY,
    "content-type": "application/json",
  },
  body: JSON.stringify({ input_text: inputText }),
});

console.log("ESTIMATE");
console.log(await estimate.text());

const submit = await fetch(`${API_BASE}/api/v1/batch-jobs`, {
  method: "POST",
  headers: {
    "x-api-key": API_KEY,
    "content-type": "application/json",
  },
  body: JSON.stringify({
    source_app: "github_example",
    input_text: inputText,
  }),
});

console.log("SUBMIT");
console.log(await submit.text());
